source/configuration/modules/mmkubernetes.rst

```diff
@@ -0,0 +1,357 @@
+
+.. _filenamerules:
+
+filenamerules
+^^^^^^^^^^^^^
+
+.. csv-table::
+   :header: "type", "default", "mandatory", "|FmtObsoleteName| directive"
+   :widths: auto
+   :class: parameter-table
+
+   "word", "SEE BELOW", "no", "none"
+
+.. note::
+
+    This directive is not supported with liblognorm 2.0.2 and earlier.
+
+When processing json-file logs, these are the lognorm rules to use to
+match the filename and extract metadata.  The default value is::
+
+    rule=:/var/log/containers/%pod_name:char-to:.%.%container_hash:char-to:_%_%names\
+    pace_name:char-to:_%_%container_name:char-to:-%-%container_id:char-to:.%.log
+    rule=:/var/log/containers/%pod_name:char-to:_%_%namespace_name:char-to:_%_%conta\
+    iner_name:char-to:-%-%container_id:char-to:.%.log
+
+.. note::
+
+    In the above rules, the slashes ``\`` ending each line indicate
+    line wrapping - they are not part of the rule.
+
+There are two rules because the `container_hash` is optional.
+
+.. _filenamerulebase:
+
+filenamerulebase
+^^^^^^^^^^^^^^^^
+
+.. csv-table::
+   :header: "type", "default", "mandatory", "|FmtObsoleteName| directive"
+   :widths: auto
+   :class: parameter-table
+
+   "word", "/etc/rsyslog.d/k8s_filename.rulebase", "no", "none"
+
+When processing json-file logs, this is the rulebase used to
+match the filename and extract metadata.  For the actual rules, see
+below `filenamerules`.
+
+.. _containerrules:
+
+containerrules
+^^^^^^^^^^^^^^
+
+.. csv-table::
+   :header: "type", "default", "mandatory", "|FmtObsoleteName| directive"
+   :widths: auto
+   :class: parameter-table
+
+   "word", "SEE BELOW", "no", "none"
+
+.. note::
+
+    This directive is not supported with liblognorm 2.0.2 and earlier.
+
+For journald logs, there must be a message property `CONTAINER_NAME`
+which has a value matching these rules specified by this parameter.
+The default value is::
+
+    rule=:%k8s_prefix:char-to:_%_%container_name:char-to:.%.%container_hash:char-to:\
+    _%_%pod_name:char-to:_%_%namespace_name:char-to:_%_%not_used_1:char-to:_%_%not_u\
+    sed_2:rest%
+    rule=:%k8s_prefix:char-to:_%_%container_name:char-to:_%_%pod_name:char-to:_%_%na\
+    mespace_name:char-to:_%_%not_used_1:char-to:_%_%not_used_2:rest%
+
+.. note::
+
+    In the above rules, the slashes ``\`` ending each line indicate
+    line wrapping - they are not part of the rule.
+
+There are two rules because the `container_hash` is optional.
+
+.. _containerrulebase:
+
+containerrulebase
+^^^^^^^^^^^^^^^^^
+
+.. csv-table::
+   :header: "type", "default", "mandatory", "|FmtObsoleteName| directive"
+   :widths: auto
+   :class: parameter-table
+
+   "word", "/etc/rsyslog.d/k8s_container_name.rulebase", "no", "none"
+
+When processing json-file logs, this is the rulebase used to
+match the CONTAINER_NAME property value and extract metadata.  For the
+actual rules, see `containerrules`.
+
+Example
+-------
+
+If you have a file `/etc/rsyslog.d/mmk8s.conf` which looks like this:
```

Comment:
Perhaps instead, something like: "An example configuration of rsyslog configured to read from docker container log file at `/var/log/containers/*.log`, processing them with the `mmkubernetes` module to add metadata, outputting JSON logs in `/var/log/k8s.log`."

This example will handle logs flowing from `journald`, but since `imjournal` is not explicitly listed, it seems out of place to mention that.